wp_all_import_functions.php

```diff
@@ -1,6 +1,103 @@
 <?php
 
-add_action('pmxi_saved_post','post_saved',5,2);
+add_action('rest_api_init', function () {
+    register_rest_route('update_broken_links/v1', '/data/', array(
+        'methods' => 'POST',
+        'callback' => 'update_broken_links_callback',
+    ));
+});
+
+function update_broken_links_callback(WP_REST_Request $request) {
+    $data = $request->get_json_params();
+
+    if (isset($data['pid']) && isset($data['spid']) && $data['mime_type']) {
+        $pids = $data['pid'];
+        $spid = $data['spid'];
+        $mime_type = $data['mime_type'];
+
+        $logs = [];
+        foreach($pids as $pid) {
+             $info = update_post_links( $pid, $spid[0], $mime_type);
+             $logs[] = implode(' ', $info);
+        }
+
+        return new WP_REST_Response(['status' => 'success', 'logs' => $logs, 'data' => $data], 200);
+    } else {
+        return new WP_REST_Response(['status' => 'error', 'message' => 'Missing params'], 500);
+    }
+}
+
+
+function update_post_links( $postid, $spostid, $mime_type= 'pdf'):array {
+    $content_post = get_post( $postid );
+    $content = $content_post->post_content;
+    $content = apply_filters( 'the_content', $content );
+
+    if ($mime_type === 'pdf') {
+        $attachments = get_attached_media( 'application/pdf', $spostid );
+        preg_match_all( '@href="([^"]+\.pdf|PDF)"@' , $content, $match_pdf );
+        $media_files = array_pop( $match_pdf );
+
+        // check for single quote pdf files.
+        preg_match_all( "@href='([^']+\.pdf|PDF)'@" , $content, $single_quote_match_pdf );
+        $single_quote_media = array_pop( $single_quote_match_pdf );
+        $media_files = array_merge($media_files, $single_quote_media);
+    }
+
+    if ($mime_type === 'img') {
+        $attachments = get_attached_media( 'image', $spostid );
+        preg_match_all( '@src="([^"]+)"@' , $content, $match_img );
+        $media_files = array_pop( $match_img );
+        // check for single quote images
+        preg_match_all( "@src='([^']+)'@" , $content, $single_quote_match_img );
+        $single_quote_media = array_pop( $single_quote_match_img );
+        $media_files = array_merge($media_files, $single_quote_media);
+    }
+
+    $update_status = 0;
+
+    // Fix broken url in Post content.
+    foreach ( $media_files as $media_file ) {
+        // If media file belongs to P4 stateless bucket, skip it.
+        if (strpos($media_file,"planet4-usa-stateless")) {
```

Comment:
If we rerun the rest API with the same param, this condition will skip already updated media.